libbeat/tests/system/base.py

```diff
@@ -17,3 +21,115 @@ def setUpClass(self):
             "winlogbeat"
         ]
         super(BaseTest, self).setUpClass()
+
+    def clean(self):
+
+        # Delete existing indices and aliases
+        try:
+            self.es.transport.perform_request('DELETE', "/*")
+        except:
+            pass
+
+        # Delete templates
+        try:
+            self.es.transport.perform_request('DELETE', "/_template/*")
+        except:
+            pass
+
+        # Delete any existing policy
+        policies = self.es.transport.perform_request('GET', "/_ilm/policy")
+        for policy, _ in policies.items():
+            try:
+                self.es.transport.perform_request('DELETE', "/_ilm/policy/" + policy)
+            except:
+                pass
+
+
+class WithESTest(BaseTest):
+    """
+    Basis for integration test with Elasticsearch
+    """
+
+    def setUp(self):
+        super(WithESTest, self).setUp()
+
+        self.elasticsearch_url = self.get_elasticsearch_url()
+        print("Using elasticsearch: {}".format(self.elasticsearch_url))
+        self.es = Elasticsearch([self.elasticsearch_url])
+        logging.getLogger("urllib3").setLevel(logging.WARNING)
+        logging.getLogger("elasticsearch").setLevel(logging.ERROR)
+
+
+class IndexAssertions(WithESTest):
```

Comment:
In general favour composition over inheritance.

How about?

```
class BaseTest(TestCase):
    @property
    def es(self):
        if self._es:
            return self._es
        ... # init es
        client = elasticsearch.Elasticsearch(...)
        return self._es

    # stolen from filebeat/tests/system/test_base.py
    @property 
    def logs(self):
        return self.log_access()

    # stolen from filebeat/tests/system/test_base.py
    def log_access(self, file=None):
        file = file if file else self.beat_name + ".log"
        return LogState(os.path.join(self.working_dir, file))

# maybe have this in a separate py file for import?
class IndexAsserts(Object):
    def __init__(self, logs, client):
        self._logs = logs
        self._client = client

    def template_not_loaded(self, template):
        ...

    def template_loaded(self, template):
        ...

    def log_contains_policy(self, policy):
        assert self._log.contains('ILM policy successfully loaded.')
        assert self._log.contains(policy)
        assert self._log.contains('"max_age": "30d"')
        assert self._log.contains('"max_size": "50gb"')

    ...

...

class TestRunTemplate(BaseTest):
    ...

    def test_template_disabled(self):
        self.renderConfig()

        proc = self.start_beat(extra_args=["-E", "setup.template.enabled=false"])
        self.wait_until(self.logs.check("mockbeat start running."))
        self.wait_until(self.logs.check("PublishEvents: 1 events have been published"))
        proc.check_kill_and_wait()

        index = IndexAsserts(self.logs, self.es)
        index.template_not_loaded(self.index_name)
```

Do we need the index assertions all in test_base? Is it possible to add a number of checks in the file they are used only?